middle_end/flambda_utils.ml

```diff
@@ -193,7 +190,7 @@ and same_named (named1 : Flambda.named) (named2 : Flambda.named) =
   | Move_within_set_of_closures _, _ | _, Move_within_set_of_closures _ ->
     false
   | Prim (p1, al1, _), Prim (p2, al2, _) ->
-    p1 = p2 && Misc.Stdlib.List.equal Variable.equal al1 al2
+    Stdlib.(=) p1 p2 && Misc.Stdlib.List.equal Variable.equal al1 al2
```

Comment:
What is the type of `p1` and `p2` here ?

Reply:
The type is `Lambda.primitive`, which has more than 100 constructors.
I reckon something should be done here; either write an equality function
for `Lambda.primitive` (and the type it references), or at the very least
ensure that the type does not change over time. However, one can
genuinely be worried that one day a float value might pop up in the
definition of `Lambda.primitive` (or one of its numerous referenced types).

Reply:
(For the record, and the impatient reader, this is the only place
where I kept the polymorphic comparison.)

Reply:
Perhaps it'd be worth adding a `Lambda.primitive_equal` function (initially an alias for `Stdlib.(=)`) to keep the definition of equality close to the definition of the type.

Reply:
@yallop Sure; if memory serves I put all introduced equality functions
in the same modules as the type definitions.

Reply:
Maybe it might be worth adding a comment to `Lambda.equal_primitive` to know why it uses polymorphic equality.

Reply:
Indeed; should I say that I am too lazy to write the equality
function because there are too many constructors and nested
types involved?

Reply:
Sure, it's a totally valid comment ^^